bin/templates/cordova/lib/prepare.js

```diff
@@ -92,6 +76,76 @@ module.exports.prepare = function (cordovaProject, options) {
     });
 };
 
+function updateUserProjectGradleConfig (configXml, defaultGradleConfigPath, projectGradleConfigPath) {
+    const defaultGradleConfig = fs.readJSONSync(defaultGradleConfigPath);
+    const profileGradleConfig = fs.readJSONSync(projectGradleConfigPath);
+
+    // Replace modified configs with defaults
+    const mergedConfigs = Object.assign(profileGradleConfig, defaultGradleConfig);
```

Comment:
The only key that can be added and is not listed in the defaults is `MAX_SDK_VERSION`.

Defining any random key is not a part of the use case, I beleive.

All default keys and `MAX_SDK_VERSION` has a reference somewhere in the Gradle build process.

I remember this mergeing process was to ensure that if the user deleted one of the default keys by mistake or removed their previous overriding setting, it would reapply the original default key & value to ensure that the build process wont fail. Only the `MAX_SDK_VERSION` key would remain deleted, if they decided to remove this key, as we do not want to enforce a max SDK version.